smarts/env/multi_scenario_v0_env.py

```diff
@@ -0,0 +1,307 @@
+        img_meters (int): Ground square size covered by image observations.
+            Defaults to 64 x 64 meter (height x width) square.
+        img_pixels (int): Pixels representing the square image observations.
+            Defaults to 256 x 256 pixels (height x width) square.
+        action_space: Action space used. Defaults to "Continuous".
+        headless (bool, optional): If True, disables visualization in
+            Envision. Defaults to False.
+        visdom (bool, optional): If True, enables visualization of observed
+            RGB images in Visdom. Defaults to False.
+        sumo_headless (bool, optional): If True, disables visualization in
+            SUMO GUI. Defaults to True.
+        envision_record_data_replay_path (Optional[str], optional):
+            Envision's data replay output directory. Defaults to None.
+
+    Returns:
+        An environment described by the input argument `scenario`.
+    """
+
+    env_specs = get_env_specs(scenario)
+    sstudio.build_scenario(scenario=[env_specs["scenario"]])
+    done_criteria = DoneCriteria(
+        collision=True,
+        off_road=True,
+        off_route=False,
+        on_shoulder=False,
+        wrong_way=False,
+        not_moving=False,
+        agents_alive=None,
+    )
+    max_episode_steps = 800
+    neighbor_radius = 50
+    road_waypoint_horizon = 50
+    waypoints_lookahead = 50
+    agent_specs = {
+        f"Agent_{i}": AgentSpec(
+            interface=AgentInterface(
+                accelerometer=True,
+                action=ActionSpaceType[action_space],
+                done_criteria=done_criteria,
+                drivable_area_grid_map=DrivableAreaGridMap(
+                    width=img_pixels,
+                    height=img_pixels,
+                    resolution=img_meters / img_pixels,
+                ),
+                lidar=True,
+                max_episode_steps=max_episode_steps,
+                neighborhood_vehicles=NeighborhoodVehicles(neighbor_radius),
+                ogm=OGM(
+                    width=img_pixels,
+                    height=img_pixels,
+                    resolution=img_meters / img_pixels,
+                ),
+                rgb=RGB(
+                    width=img_pixels,
+                    height=img_pixels,
+                    resolution=img_meters / img_pixels,
+                ),
+                road_waypoints=RoadWaypoints(horizon=road_waypoint_horizon),
+                waypoints=Waypoints(lookahead=waypoints_lookahead),
+            ),
+        )
+        for i in range(env_specs["num_agent"])
+    }
+
+    env = HiWayEnv(
+        scenarios=[env_specs["scenario"]],
+        agent_specs=agent_specs,
+        sim_name="MultiScenario",
+        headless=headless,
+        visdom=visdom,
+        sumo_headless=sumo_headless,
+        envision_record_data_replay_path=envision_record_data_replay_path,
+    )
+    env = _InfoScore(env=env)
+
+    return env
+
+
+def get_env_specs(scenario: str):
```

Comment:
This method makes me slightly nervous to leave public since it is used under the hood in the environment construction without the association being quite clear.

```suggestion
def _get_env_specs(scenario: str):
```

Reply:
Changed it to `_get_env_specs()`.